cmp/grid/Grid.js

```diff
@@ -387,28 +387,32 @@ class LocalModel {
                     prevCount = prevRs ? prevRs.count : 0;
 
                 withShortDebug(`${isUpdate ? 'Updated' : 'Loaded'} Grid`, () => {
-                    if (prevCount !== 0 && experimental.useTransactions) {
-                        const transaction = this.genTransaction(newRs, prevRs);
+                    let transaction = null;
+                    if (prevCount !== 0) {
+                        transaction = this.genTransaction(newRs, prevRs);
                         console.debug(this.transactionLogStr(transaction));
 
                         if (!this.transactionIsEmpty(transaction)) {
                             agApi.applyTransaction(transaction);
                         }
+
                     } else {
                         agApi.setRowData(newRs.list);
                     }
 
                     if (experimental.externalSort) {
-                        const {sortBy} = model;
-                        if (!isEqual(sortBy, this._lastSortBy)) agGridModel.applySortBy(sortBy);
-                        this._lastSortBy = sortBy;
+                        agGridModel.applySortBy(model.sortBy);
                     }
 
                     this.updatePinnedSummaryRowData();
 
-                    const refreshCols = model.columns.filter(c => !c.hidden && c.rendererIsComplex);
-                    if (!isEmpty(refreshCols)) {
-                        agApi.refreshCells({columns: refreshCols.map(c => c.colId), force: true});
+                    if (!isEmpty(transaction?.update)) {
+                        const refreshCols = model.columns.filter(c => !c.hidden && c.rendererIsComplex);
+                        if (refreshCols) {
+                            const rowNodes = compact(transaction.update.map(r => agApi.getRowNode(r.id))),
```

Comment:
Nice optimization here! Is a place I had noted as something we could be more efficient with when I was looking into inline editing performance, though there is a caveat here - if you have cell renderers which depend on other records in the store they will no longer be refreshed when the records they may depend on are updated. Something we might want to note in the changelog if we think there may be apps which depend on that kind of behavior.

Reply:
I think the documented semantics are clear that `rendererIsComplex` supported dependencies on something else on the *row*.  Do you think that is not enough power?   Do we need a `rendererIsReallyComplex` property  ?   


=============================
true if this renderer relies on more than
     *      just the value of the field associated with this column.  Set to true to ensure that
     *      the cells for this column are updated any time the record is changed.  Setting to true
     *      may have performance implications. Default false.

Reply:
Ok yeah I think those docs are pretty clear that it is only refreshed when the record itself changes - but if an app was depending on other records it would have worked previously, but now it wont - so maybe just something for us to be aware of in case some renderers stop working?

Something like a `refreshMode` for Columns might be worth considering for more advanced use cases